### DISCUSSION TITLE
Suddenly no CSP violations when using styles

### DISCUSSION THREAD

**Discussion by sr258** [lit/lit#3271] [Q&A]
We use Lit components that import styles via ES6 `import` statements. As we need to set a pretty strict content security policy, we had to generate hashes for the styles and add these hashes to the CSP like this: `style-src 'self' 'sha256-XXXXXXXXXXXX'`

As of late, we suddenly don't need the hashes anymore and only setting style-src 'self' works. Was there a change in how lit loads styles that allows us to get rid of using hashes? We would greatly welcome this, but I haven't found any documentation on a change.

**Comment by justinfagnani** [comment]
> We use Lit components that import styles via ES6 import statements

Can you give more detail here?

- What browsers are you running in?
- What specific kind of "CSS imports" are you using?

 Are you importing native CSS modules with import assertions?

```ts
import styles from './styles.css' assert {type: 'css};
```

Or importing Lit CSS strings:

```ts
export const styles = css`:host {display: block;}`;
```

```
import {styles} from './styles.js';
```

_or_ importing non-standard Webpack-style CSS modules:

```ts
improt {classOne} from './styles.css';
```

**Reply by sr258** [reply]
Sorry about answering so super late. Turned out the violations only happen in Safari browsers at the moment and all other browsers don't have these violations. I believe this was different in the past, but at least I have an explanation.
